lib/datadog/core/configuration/option.rb

```diff
@@ -94,6 +96,91 @@ def default_precedence?
 
         private
 
+        def coerce_env_variable(value)
```

Comment:
I believe the intention here is to replace `Datadog::Core::Environment::VariableHelpers`. Should we go ahead and mark all those methods as deprecated for removal?

Reply:
I agree with that. There is still one setting that still uses `Datadog::Core::Environment::VariableHelpers`

```ruby
settings :client_ip do
  # Whether client IP collection is enabled. When enabled client IPs from HTTP requests will
  #   be reported in traces.
  #
  # Usage of the DD_TRACE_CLIENT_IP_HEADER_DISABLED environment variable is deprecated.
  #
  # @see https://docs.datadoghq.com/tracing/configure_data_security#configuring-a-client-ip-header
  #
  # @default `DD_TRACE_CLIENT_IP_ENABLED` environment variable, otherwise `false`.
  # @return [Boolean]
  option :enabled do |o|
    o.type :bool
    o.default do
      disabled = env_to_bool(Tracing::Configuration::Ext::ClientIp::ENV_DISABLED)

      enabled = if disabled.nil?
                  false
                else
                  Datadog::Core.log_deprecation do
                    "#{Tracing::Configuration::Ext::ClientIp::ENV_DISABLED} environment variable is deprecated, use #{Tracing::Configuration::Ext::ClientIp::ENV_ENABLED} instead."
                  end

                  !disabled
                end

      # ENABLED env var takes precedence over deprecated DISABLED
      env_to_bool(Tracing::Configuration::Ext::ClientIp::ENV_ENABLED, enabled)
    end
  end
end
```

I guess it is ok to mark the methods from `Datadog::Core::Environment::VariableHelpers` as deprecated. 

I would like to that on separate PR to avoid touching more parts of the codebase on this single PR